File: MaaS-go/api-gateway/internal/router/router.go
package router

import (
	"github.com/gin-gonic/gin"

	"maas-platform/api-gateway/internal/handler"
)

// RegisterRoutes registers all routes
func RegisterRoutes(r *gin.RouterGroup, h *handler.Handler) {
	// Auth routes (no authentication required)
	auth := r.Group("/auth")
	{
		auth.POST("/login", h.Login)
		auth.POST("/register", h.Register)
	}

	// Protected routes
	protected := r.Group("")
	{
		// User routes
		users := protected.Group("/users")
		{
			users.GET("/me", h.GetCurrentUser)
		}

		// Model routes
		models := protected.Group("/models")
		{
			models.POST("", h.CreateModel)
			models.GET("", h.ListModels)
			models.GET("/:id", h.GetModel)
			models.PUT("/:id", h.UpdateModel)
			models.DELETE("/:id", h.DeleteModel)
			models.PATCH("/:id/status", h.UpdateModelStatus)
			models.POST("/:id/tags", h.AddModelTags)
			models.DELETE("/:id/tags", h.RemoveModelTags)
			models.GET("/:id/metadata", h.GetModelMetadata)
			models.PUT("/:id/metadata", h.SetModelMetadata)
		}

		// Inference routes
		protected.POST("/inference", h.RunInference)
	}
}

File: MasS-python/api_gateway/internal/router/router.py
from fastapi import APIRouter, FastAPI

from api_gateway.internal.handler import health_router, model_router, user_router
from api_gateway.internal.handler.config_handler import router as config_router


def register_routes(app: FastAPI) -> None:
    app.include_router(health_router)
    app.include_router(config_router)

    api_v1 = APIRouter(prefix="/api/v1")
    api_v1.include_router(user_router, prefix="/users", tags=["users"])
    api_v1.include_router(model_router, prefix="/models", tags=["models"])
    app.include_router(api_v1)
